### PR TITLE
Update web3 1.0.0-beta.48, add timeout for checking transaction to be 1 minute

### PR DESCRIPTION
**Pull request by manhlx3006**

### CODE REVIEW

File: src/sagas/transactionSaga.js [hunk 1]
```diff
@@ -18,16 +18,24 @@ const getTransferState = state => state.transfer;
...
 
   while(!isTxMined) {
     const txReceipt = yield call(web3.eth.getTransactionReceipt, txHash);
 
-    if (txReceipt && txReceipt.status === '0x1') {
+    if (txReceipt && txReceipt.status) {
       yield put(txActions.setIsTxMined(txReceipt.status));
       isTxMined = true;
-    } else if (txReceipt && txReceipt.status === '0x0') {
+    } else if (txReceipt && txReceipt.status) {
```
Comment (artwooood) [inline src/sagas/transactionSaga.js]: It should be `!txReceipt.status`

File: src/sagas/transactionSaga.js [hunk 2]
```diff
@@ -18,16 +18,24 @@ const getTransferState = state => state.transfer;
...
       yield put(txActions.setTxError("There is something wrong with the transaction!"));
       isTxMined = true;
+    } else {
+      let currentTime = Date.now(); // current time
+      if (currentTime - startTime >= appConfig.TRANSACTION_TIME_OUT) {
```
Comment (artwooood) [inline src/sagas/transactionSaga.js]: Doesn't it always return negative number? To subtract 2 `Date` we should use `Math.abs`

**Review by artwooood — CHANGES_REQUESTED** [review]

File: src/sagas/transactionSaga.js (hunk 2 above)
Comment (manhlx3006) [reply src/sagas/transactionSaga.js]: It will always return positive number, but to be safer, I'll use `Math.abs` as u suggested

Comment (artwooood) [reply src/sagas/transactionSaga.js]: Ah you take `currentTime - StartTime`, thought it was `StartTime - currentTime` ✌️

**Review by artwooood — APPROVED** [review]

**Pull request merged**